pkg/server/context.go

```diff
@@ -38,6 +45,16 @@ func NewContext(ctx context.Context, cfg string) (*Context, error) {
 		return nil, err
 	}
 
+	// Construct a custom user-agent string for the apply client used by the deploy controller
+	// so that we can track which node's deploy controller most recently modified a resource.
+	nodeName := os.Getenv("NODE_NAME")
+	managerName := deploy.ControllerName + "@" + nodeName
+	if nodeName == "" || len(managerName) > validation.FieldManagerMaxLength {
```

Comment:
would we want to truncate somehow, such as with wrangler's name.SafeName() function?

Reply:
I figured truncating is useless, since we can't use the truncated node name to find the corresponding Node object, and I'm hoping to piggyback off this to have the helm controller set a preferred node for the job pod if one can be found.